version: '3'
services:
  django:
    restart: always
    image: artsiom24091/django:latest
    build: .
    container_name: django
    ports:
      - "80:8000"
      - "8080:8080"
    environment:
      SECRET_KEY: "django-insecure-osfh^r%nvpun6ek2z4n!f*&r@$g+*d8n#8z0ac9_%)%api0x)7"
      DEBUG: "True"
    volumes:
      -  /home/server/reps/images:/usr/src/app/images
      -  /home/server/reps/database:/usr/src/app/database
  face_recognition_queue:
    restart: always
    image: artsiom24091/face_recognition_queue:latest
    build: .
    container_name: face_recognition_queue
    volumes:
      -  /home/server/reps/images:/usr/src/app/images
      -  /home/server/reps/database/dataset:/usr/src/app/dataset
    tty: true
    network_mode: host
  proxy:
    restart: always
    image: artsiom24091/proxy:latest
    build: .
    container_name: proxy
    ports:
      - "3066:3066"
  safety_tracker:
    restart: always
    image: artsiom24091/safety_tracker:latest
    build: .
    container_name: safety_tracker
    ports:
      - "3020:3020"
    depends_on:
      - django
    environment:
      CAMERA_URLS: 'rtsp://admin:just4Taqtile@192.168.1.160/h264_stream'
    volumes:
      -  /home/server/reps/images:/var/www/5scontrol/images
      -  /home/server/reps/database/dataset:/var/www/5scontrol/dataset
  5scontrol_front:
    restart: always
    image: artsiom24091/5scontrol_front:latest
    build: .
    container_name: 5scontrol_front
    ports:
      - "3000:3000"
    depends_on:
      - django
    tty: true
    network_mode: host
